Proposal for terminology update (#75)
ietf-ivy-wg/network-inventory-yang: draft-ietf-ivy-network-inventory-yang.md

## Terminology

The following terms are defined in {{!RFC7950}} and are not redefined here:

> - client

Reviewer: Not used in the doc

Author: ```suggestion
```

- server
- augment
- data model

> - data node

Reviewer: check if this is used.

Author: - server is used
- augment is used in derivate forms (to augment, augmentation)
- data model
- data node is used in the comments within YANG model and it will be used in the Security Considerations (based on RFC 8407-bis template)

The following terms are defined in {{!RFC6241}} and are not redefined here:

> - configuration data

Reviewer: This is not used in the doc.

Author: Now used by the text added in https://github.com/ietf-ivy-wg/network-inventory-yang/pull/75/commits/d493a0e77dd8f728bd0e69f3e5ed40c4ea11aa7a to clarify that the scope is only to report actual inventory data and not to configure planned/intended inventory data, as per mailing list discussion:

https://mailarchive.ietf.org/arch/msg/inventory-yang/-9MNndP4ykQSPIMSCgup0VI9wTo/

- state data

> The following terms are defined in {{IANA_YANG}} and are not redefined here:

Reviewer: {{IANA_YANG}} does not have these terms!

I guess you meant [RFC 8348](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwi4w_z21aSKAxU_UaQEHQuRC98QFnoECBYQAQ&url=https%3A%2F%2Fdatatracker.ietf.org%2Fdoc%2Fhtml%2Frfc8348&usg=AOvVaw37XEzeeG-tUB3JYKmiSmSS&opi=89978449)

Author: My understanding is that these terms are defined in the description statements of the component classes identities within [IANA_YANG]

```suggestion
The following terms are defined in the description statements of the corresponding YANG identities, defined in {{IANA_YANG}}, and are not redefined here:
```

Author: **[2025-01-29 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/wiki/minutes-2025-01-29)**

Agreed the proposed resolution in https://github.com/ietf-ivy-wg/network-inventory-yang/pull/75/files#r1886557028

> - backplane

Reviewer: All these terms are not formally defined in 8348. Maybe change to "used" or "introduced".

Author: See: https://github.com/ietf-ivy-wg/network-inventory-yang/pull/75/files#r1886557028

- battery
- container
- cpu
- chassis
- fan
- module
- port
- power supply
- sensor
- stack
- storage device

> Note that the definition of port component in {{IANA_YANG}} needs to be refined in future version of this document.

> > TBD: Recap the concept of chassis/slot/component/board/... in {{TMF_SD2-20}}.

Reviewer 2: I think it would be better to create a related issue about that instead to have this note in the draft. Just a suggestion.

Author: This should be resolved by the appendix which is work-in-progress in PR #66

Also, the document makes use of the following terms:

Network Inventory:
: A collection of data for network elements and their components managed by a specific management system.

Physical Network Element:

> : An implementation or application specific groupings of components (e.g., hardware components).

Author: ```suggestion
: An implementation or application specific group of components (e.g., hardware components).
```

Network Element:

> : The generalization of the physical network element definition to include other types of inventory objects which can be managed, from an inventory perspective, like physical network elements.

Reviewer 2: If you "include other types of inventory objects" you probably point to "other components".
Why do not say "The generalization of the physical network element definition to include other types of inventory objects which can be managed, from an inventory perspective, like components of physical network elements."

Author: ```suggestion
: The generalization of the physical network element definition.
```

Hardware Component:

> : The generalization of the hardware components defined in {{IANA_YANG}} (e.g., backplane, battery, container, cpu, chassis, fan, module, port, power supply, sensor, stack and storage device components).

Reviewer: ```suggestion
: The generalization of the hardware components defined in {{IANA_YANG}} (e.g., backplane, battery, container, cpu, chassis, fan, module, port, power supply, sensor, stack, and storage device components).
```

Author: **[2025-01-29 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/wiki/minutes-2025-01-29)**

Proposal agreed

> : The list of hardware components can be extended in future versions of {{IANA_YANG}}.

Reviewer: We can't formally directly add those the IANA module. These should be added to IANA-ENTITY-MIB:

```
>    The "iana-hardware" YANG module is intended to reflect the
>    "IANA-ENTITY-MIB" MIB module so that if a new enumeration is added to
>    the "IANAPhysicalClass" textual convention, the same class is added
>    as an identity derived from "ianahw:hardware-class".
```

Author: ```suggestion
: The list of hardware components can be extended in future versions of {{IANA_ENTITY_MIB}} (and, consequently, of ({{IANA_YANG}}).
```

Looking at IANA web, I think the {{IANA_ENTITY_MIB}} can reference the following URL:

https://www.iana.org/assignments/ianaentity-mib/ianaentity-mib.xhtml

I am also wondering whether the {{IANA_YANG}} should be changed to {{IANA_HW_YANG}} and reference the following URL:

https://www.iana.org/assignments/iana-hardware/iana-hardware.xhtml

Author: ```suggestion
: The list of hardware components can be extended in future versions of {{IANA_ENTITY_MIB}} (and, as a consequence, of ({{IANA_YANG}}).
```

Component:
: The generalization of the hardware component definition to include other inventory objects which can be managed, from an inventory perspective, like hardware components.

Slot:
: A holder of the board.

Board/Card:

> : A pluggable equipment can be inserted into one or several slots (or sub-slots) and can afford a specific transmission function independently.

Reviewer 3: My thoughts and Sergio's comments labeled with [SB]

The issue is finding terminology that will be agreeable to network equipment vendors, data center people, and computer hardware people.  The idea is simple, but can be mapped onto different terminology.

For example.

A rack is a metal frame that may or may not contain electrical components that holds a chassis.

A standard rack holding something like a Cisco 8804 chassis looks like this.

![download](https://github.com/user-attachments/assets/f52983c7-d6dd-4e5a-8da5-716a02daa1aa)

The big holes you see in the picture above are slots.  There are slots on the back as well.  A slot can accept various modules, trays, or cards.  Some modules and/or cards can have daughter cards can be connected to the module or card.

 

Looking at the Network Inventory YANG terminology.

The rack, chassis and slots above are (as TMF puts them) holders. 

[SB>] YES the TMF separation between “holders” and “equipment” groups  is also in our document

An example of a sub-chassis could be a tray that holds other components (like fans).  In the Network Inventory YANG the term Board is (I think) a module or a card. 

[SB>] In our model a board is defined as” ianahw:module” , there is no distinction unfortunately from module to card/board.

Usually, modules are things like fans or power, so if we are modeling that we will need to differentiate a module from a card (so calling both of those board is probably wrong). 

[SB>]  Fan is considered as a specific card, it has a definition specific, not as ianahw:module .

There is a differentiation between cards as well (at least in Cisco-land).  There are “line cards” and “fabric cards”.  A line card has ports on it and in the case of optical pluggables a line card may have slots to accept pluggables. I don’t know if that is what is intended by the “sub-slot” terminology, but it could work since a pluggable would be considered a module that fits into a cards' sub-slot.  At least that is one interpretation. 

[SB>] In this case we would have a definition for sub-slot that should be “a slot of a line-card hosting an optical pluggable”

A “fabric card” is a card that “line cards” and sometimes modules plug into.  But here the "sub-slot" terminology doesn't really work very well, because the "sub-slot" on the "fabric card" is not inline with the slot the "fabric card" is in.  

What needs to be made clear is whether or not a “sub-slot” is simply a holder or if a “sub-slot” can be used for something like an optical pluggable.

[SB>] It can be used as a holder for an optical pluggable in my view.

I think a couple of examples to see what the instance data would look would help.

Looking at M.3100.  I would say recursive not circular.

 

A rack is made up of bays

A chassis is held in some number of bays of a rack depending on the size of the chassis.

A chassis has slots (or physical places to put things)

A chassis slot holds a "circuit pack" (which can be a line card, fabric card, computer, power, fans etc.)

A line card can also accept pluggables

 

I think leveraging M.3100 and mapping the terminology is a good exercise.  Things will get more interesting when the data center people show up and want to start talking about blades.

> : A pluggable equipment can be inserted into one or several slots (or sub-slots) and can afford a specific transmission function independently.

Author: **[2025-01-22 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/wiki/minutes-2025-01-22)**

- [ ] @nigel-r-davis to check TMF supporting documents as well as the comments from Scott and Sergio in https://github.com/ietf-ivy-wg/network-inventory-yang/pull/75#discussion_r1909574652

> : A pluggable equipment can be inserted into one or several slots (or sub-slots) and can afford a specific transmission function independently.

Author: **[2025-01-29 Base Network Inventory weekly call](https://github.com/ietf-ivy-wg/network-inventory-yang/wiki/minutes-2025-01-29)**

> * [ ]  @nigel-r-davis to check TMF supporting documents as well as the comments above from Scott and Sergio

Not good definition found so far: investigation is continuing

New issue #79 has been created to track this open issue.

# YANG Data Model for Network Inventory Overview

The network element definition is generalized to support physical

> devices and other types of inventory objects that can be managed as physical network elements from an

Author: ```suggestion
 network elements and other types of components' groups that can be managed as physical network elements from an
```

> devices and other types of inventory objects that can be managed as physical network elements from an
> inventory perspective.

Author: ```suggestion
 network elements and other types of components' groups that can be managed as physical network elements from an
inventory perspective.

Physical network elements are usually devices such as hosts, gateways, terminal servers, and the like, which have management agents responsible for performing the network management functions requested by the network management stations ({?RFC1157}).
```
